### PR TITLE
Support the function OBJECT_DEFINITION()

### PR DESCRIPTION
**Pull request by shalinilohia50**
### Description
Implement a function OBJECT_DEFINITION(object_id) which returns the Transact-SQL source text of the definition of a specified object.

List of objects supported:

1. Check constraints
2. Default constraints
3. Procedure
4. Scalar function
5. DML trigger - TODO: After BABEL-3927 is fixed, we get the correct trigger definition.
6. Views
7. Inline table valued functions
8. Table valued functions

Task: BABEL-3919
Signed-off-by: Shalini Lohia <lshalini@amazon.com>

### Issues Resolved
BABEL-3919

### Test Scenarios Covered ###
* **Use case based -**
Added use case based tests in JDBC framework

* **Boundary conditions -**
Added edge cases where the object_id is either incorrect or NULL

* **Arbitrary inputs -**
* **Negative test cases -**
* **Minor version upgrade tests -**
* **Major version upgrade tests -**
* **Performance tests -**
* **Tooling impact -**
* **Client tests -**

### CODE REVIEW

**Review by rishabhtanwar29 — DISMISSED** [review]
We should update the description with the object types which are covered with the current implementation.

File: contrib/babelfishpg_tsql/sql/sys_functions.sql [hunk 1]
```diff
@@ -3377,3 +3377,30 @@ $$
     SELECT current_setting('application_name');
 $$
 LANGUAGE sql PARALLEL SAFE STABLE;
+
+CREATE OR REPLACE FUNCTION OBJECT_DEFINITION(IN object_id INT)
```
Comment (HarshLunagariya) [inline contrib/babelfishpg_tsql/sql/sys_functions.sql:3381]: Is it supposed to work for cross-db calls also? If yes then using sys views won't be sufficient.

**Review by rishabhtanwar29 — DISMISSED** [review]

File: contrib/babelfishpg_tsql/sql/sys_functions.sql [hunk 2]
```diff
@@ -3377,3 +3377,30 @@ $$
...
+            IF (definition IS NULL)
+            THEN
+                RETURN NULL;
+            END IF;
+        END IF;
+    END IF;
+
+    RETURN definition;
+END;
+$$
```
Comment (HarshLunagariya) [inline contrib/babelfishpg_tsql/sql/sys_functions.sql:3404]: It should throw error when caller doesn't have permission on object acc. documentation. Please add test cases for the same.

**Review by HarshLunagariya — CHANGES_REQUESTED** [review]

File: contrib/babelfishpg_tsql/sql/sys_functions.sql (hunk 1 above)
Comment (shalinilohia50) [reply contrib/babelfishpg_tsql/sql/sys_functions.sql:3381]: No, it should show NULL for cross-db call. Also, added test showing cross-db call.

File: contrib/babelfishpg_tsql/sql/sys_functions.sql (hunk 2 above)
Comment (shalinilohia50) [reply contrib/babelfishpg_tsql/sql/sys_functions.sql:3404]: The OBJECT_DEFINITION() function doesn't throw error in that case. I have added tests for that case, it returns NULL.
From T-SQL doc:
```
Returns NULL on error or if a caller does not have permission to view the object.

**Review by HarshLunagariya — APPROVED** [review]

**Review by rishabhtanwar29 — APPROVED** [review]

**Review by shardgupta — APPROVED** [review]

**Pull request merged**